File: src/test/java/com/flexpoker/table/command/aggregate/singlehand/twoplayer/TwoPlayerSeatPositionsAndButtonAndBlindsTest.java
package com.flexpoker.table.command.aggregate.singlehand.twoplayer;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.UUID;
import java.util.stream.Collectors;

import org.junit.jupiter.api.Test;

import com.flexpoker.table.command.aggregate.testhelpers.TableTestUtils;
import com.flexpoker.table.command.events.HandDealtEvent;
import com.flexpoker.table.command.events.TableCreatedEvent;

public class TwoPlayerSeatPositionsAndButtonAndBlindsTest {

    @Test
    void test() {
        var tableId = UUID.randomUUID();
        var player1Id = UUID.randomUUID();
        var player2Id = UUID.randomUUID();

        var table = TableTestUtils.createBasicTableAndStartHand(tableId, player1Id, player2Id);

        // check seat positions
        var seatPositionToPlayerIdMap = ((TableCreatedEvent) table.fetchNewEvents().get(0)).getSeatPositionToPlayerMap();
        seatPositionToPlayerIdMap = seatPositionToPlayerIdMap.entrySet().stream()
                .filter(x -> x.getValue() != null)
                .collect(Collectors.toMap(x -> x.getKey(), x -> x.getValue()));

        var player1MatchList = seatPositionToPlayerIdMap.values().stream()
                .filter(x -> x.equals(player1Id)).collect(Collectors.toList());
        var player2MatchList = seatPositionToPlayerIdMap.values().stream()
                .filter(x -> x.equals(player2Id)).collect(Collectors.toList());

        // verify that each player id is only in one seat position and that
        // those are the only two filled-in positions
        assertEquals(1, player1MatchList.size());
        assertEquals(1, player2MatchList.size());

        var numberOfOtherPlayerPositions = seatPositionToPlayerIdMap.values().stream()
                .filter(x -> !x.equals(player1Id)).filter(x -> !x.equals(player2Id))
                .distinct().count();
        assertEquals(0, numberOfOtherPlayerPositions);

        // check blinds
        var player1Position = seatPositionToPlayerIdMap.entrySet().stream()
                .filter(x -> x.getValue().equals(player1Id)).findAny().get().getKey()
                .intValue();
        var player2Position = seatPositionToPlayerIdMap.entrySet().stream()
                .filter(x -> x.getValue().equals(player2Id)).findAny().get().getKey()
                .intValue();

        var handDealtEvent = ((HandDealtEvent) table.fetchNewEvents().get(2));

        if (player1Position == handDealtEvent.getButtonOnPosition()) {
            assertEquals(player1Position, handDealtEvent.getButtonOnPosition());
            assertEquals(player1Position, handDealtEvent.getSmallBlindPosition());
            assertEquals(player2Position, handDealtEvent.getBigBlindPosition());
        } else if (player2Position == handDealtEvent.getButtonOnPosition()) {
            assertEquals(player2Position, handDealtEvent.getButtonOnPosition());
            assertEquals(player2Position, handDealtEvent.getSmallBlindPosition());
            assertEquals(player1Position, handDealtEvent.getBigBlindPosition());
        } else {
            throw new IllegalStateException("for a new two-player hand, one of the players must be the button");
        }
    }

}

File: src/test/kotlin/com/flexpoker/table/command/aggregate/singlehand/twoplayer/TwoPlayerSeatPositionsAndButtonAndBlindsTest.kt
package com.flexpoker.table.command.aggregate.singlehand.twoplayer

import com.flexpoker.table.command.aggregate.testhelpers.createBasicTableAndStartHand
import com.flexpoker.table.command.events.HandDealtEvent
import com.flexpoker.table.command.events.TableCreatedEvent
import com.flexpoker.test.util.UnitTestClass
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.util.UUID

@UnitTestClass
class TwoPlayerSeatPositionsAndButtonAndBlindsTest {

    @Test
    fun test() {
        val tableId = UUID.randomUUID()
        val player1Id = UUID.randomUUID()
        val player2Id = UUID.randomUUID()
        val events = createBasicTableAndStartHand(tableId, player1Id, player2Id)

        // check seat positions
        val seatPositionToPlayerIdMap = (events[0] as TableCreatedEvent)
            .seatPositionToPlayerMap.filter { it.value != null }
        val player1MatchList = seatPositionToPlayerIdMap.values.filter { it == player1Id }
        val player2MatchList = seatPositionToPlayerIdMap.values.filter { it == player2Id }

        // verify that each player id is only in one seat position and that
        // those are the only two filled-in positions
        assertEquals(1, player1MatchList.size)
        assertEquals(1, player2MatchList.size)
        val numberOfOtherPlayerPositions = seatPositionToPlayerIdMap.values
            .filter { it != player1Id }.filter { it != player2Id }.distinct().count()
        assertEquals(0, numberOfOtherPlayerPositions)

        // check blinds
        val player1Position = seatPositionToPlayerIdMap.entries.first { it.value == player1Id }.key
        val player2Position = seatPositionToPlayerIdMap.entries.first { it.value == player2Id }.key
        val (_, _, _, _, _, _, buttonOnPosition, smallBlindPosition, bigBlindPosition) = events[2] as HandDealtEvent
        when {
            player1Position == buttonOnPosition -> {
                assertEquals(player1Position, buttonOnPosition)
                assertEquals(player1Position, smallBlindPosition)
                assertEquals(player2Position, bigBlindPosition)
            }
            player2Position == buttonOnPosition -> {
                assertEquals(player2Position, buttonOnPosition)
                assertEquals(player2Position, smallBlindPosition)
                assertEquals(player1Position, bigBlindPosition)
            }
            else -> {
                throw IllegalStateException("for a new two-player hand, one of the players must be the button")
            }
        }
    }

}